Pod/Classes/PINAnimatedImageManager.m

```diff
@@ -449,16 +524,22 @@ + (NSFileHandle *)fileHandle:(NSError **)error filePath:(NSString **)filePath te
  
  */
 
-+ (void)writeFileHeader:(NSFileHandle *)fileHandle width:(UInt32)width height:(UInt32)height loopCount:(UInt32)loopCount frameCount:(UInt32)frameCount bitmapInfo:(UInt32)bitmapInfo durations:(Float32*)durations
++ (NSError *)writeFileHeader:(NSFileHandle *)fileHandle width:(UInt32)width height:(UInt32)height loopCount:(UInt32)loopCount frameCount:(UInt32)frameCount bitmapInfo:(UInt32)bitmapInfo durations:(Float32*)durations
 {
-  UInt16 version = 1;
-  [fileHandle writeData:[NSData dataWithBytes:&version length:sizeof(version)]];
-  [fileHandle writeData:[NSData dataWithBytes:&width length:sizeof(width)]];
-  [fileHandle writeData:[NSData dataWithBytes:&height length:sizeof(height)]];
-  [fileHandle writeData:[NSData dataWithBytes:&loopCount length:sizeof(loopCount)]];
-  [fileHandle writeData:[NSData dataWithBytes:&frameCount length:sizeof(frameCount)]];
-  [fileHandle writeData:[NSData dataWithBytes:&bitmapInfo length:sizeof(bitmapInfo)]];
-  [fileHandle writeData:[NSData dataWithBytes:durations length:sizeof(Float32) * frameCount]];
+  NSError *error = nil;
+  @try {
+    UInt16 version = 1;
+    [fileHandle writeData:[NSData dataWithBytes:&version length:sizeof(version)]];
+    [fileHandle writeData:[NSData dataWithBytes:&width length:sizeof(width)]];
+    [fileHandle writeData:[NSData dataWithBytes:&height length:sizeof(height)]];
+    [fileHandle writeData:[NSData dataWithBytes:&loopCount length:sizeof(loopCount)]];
+    [fileHandle writeData:[NSData dataWithBytes:&frameCount length:sizeof(frameCount)]];
+    [fileHandle writeData:[NSData dataWithBytes:&bitmapInfo length:sizeof(bitmapInfo)]];
+    [fileHandle writeData:[NSData dataWithBytes:durations length:sizeof(Float32) * frameCount]];
```

Comment:
Not sure if this is correct from the code. I would imagine that the value of durations would be used in the length calculation?